appinventor/components/src/com/google/appinventor/components/runtime/ListView.java

```diff
@@ -1361,14 +1360,22 @@ public void getItemOffsets(Rect outRect, View view, RecyclerView parent, Recycle
           outRect.setEmpty();
         }
       } else {
-        int column = position % spanCount;
-        outRect.left = margins - column * margins / spanCount;
-        outRect.right = (column + 1) * margins / spanCount;
-        if (position < spanCount || first) {
-          first = false;
-          outRect.top = margins;
+        if (orientation == ComponentConstants.LAYOUT_ORIENTATION_HORIZONTAL) {
+          if (position == 0) {
+            outRect.set(margins, margins, margins, margins);
+          } else {
+            outRect.set(0, margins, margins, margins);
```

Comment:
So one issue with doing this here is that it seems to make the elements unbalanced. Here are two screen shots with the first and last elements of the list in horizontal layout mode. Note how the last element is flush with the edge whereas the first element isn't.

![Screenshot_20241213-121310](https://github.com/user-attachments/assets/79aa213e-c341-41a0-8749-495cdc146316)

![Screenshot_20241213-121314](https://github.com/user-attachments/assets/daf39e94-f121-423e-a2e6-22cb5b3d48b1)

Reply:
Yes, I focused on margins, not paying attention to element widths. To fix this we need to set the element width. I also added a screen orientation listener to update widths after screen rotation. I also simplified the code for setting parenthesis for vertical layout, removing lines of code that I had tested with Grid layout before.